package-build.el

```diff
@@ -1613,6 +1615,13 @@ If FILE-NAME is not specified, the default archive-contents file is used."
 ;; separate package.  Note also that it would be straightforward to
 ;; generate the SVG ourselves, which would save the network overhead.
 
+(defun package-build--melpa-badge-image-version (config archive-entry)
+  "For CONFIG, determine a version label for ARCHIVE-ENTRY."
+  (when (memq (plist-get config :fetcher) '(github gitlab))
+    (let ((commit (alist-get :commit (elt (cdr archive-entry) 4))))
+      (when (and commit (stringp commit) (= 40 (length commit)))
```

Comment:
As I understand it, the MELPA server is running on an old Emacs (i.e., an emacs without `when-let`).  If theres another macro in-scope that I could use, let me know.